From b7ce908933761660417c82c4c31216e22fc61a3f Mon Sep 17 00:00:00 2001
From: wfeng6 <wei.feng@intel.com>
Date: Fri, 26 Jul 2013 17:01:37 +0800
Subject: Update the crop information before dequeue buffers from NativeWindow

BZ: 124114

In video editor application, if the 1st item is a clip and the 3rd
item is an image, the crop information will not be updated. This
causes the picture is stretched in the preview window.

In this patch, the crop information will be updated before dequeue
buffers from NativeWindow.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: I796c8f7ffde981450f7221d0c0e33369bf5a2ac4
Orig-Change-Id: I47b0bf0bb8e30dd49fdbf59038e8cb2aad459e2f
Signed-off-by: wfeng6 <wei.feng@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/lvpp/NativeWindowRenderer.cpp |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/libvideoeditor/lvpp/NativeWindowRenderer.cpp b/libvideoeditor/lvpp/NativeWindowRenderer.cpp
index e5c6df6..b02a3c6 100755
--- a/libvideoeditor/lvpp/NativeWindowRenderer.cpp
+++ b/libvideoeditor/lvpp/NativeWindowRenderer.cpp
@@ -401,6 +401,14 @@ void NativeWindowRenderer::queueExternalBuffer(ANativeWindow* anw,
     MediaBuffer* buffer, int width, int height) {
     native_window_set_buffers_geometry(anw, width, height,
             HAL_PIXEL_FORMAT_YV12);
+
+    android_native_rect_t crop;
+    crop.left = 0;
+    crop.top = 0;
+    crop.right = width - 1;
+    crop.bottom = height - 1;
+    native_window_set_crop(anw, &crop);
+
     native_window_set_usage(anw, GRALLOC_USAGE_SW_WRITE_OFTEN);
 
     ANativeWindowBuffer* anb;
-- 
1.7.4.1

